cli/flags_test.go

```diff
@@ -160,6 +160,161 @@ func TestFlagSet_Default(t *testing.T) {
 	})
 }
 
+func TestFlagSection_StringMapVar(t *testing.T) {
```

Comment:
Can you add some test cases for when someone passes a value equal to the default as the first arg and they have more than one arg? Just to confirm that it won't remove the user's first arg